models/fates/R/model2netcdf.FATES.R

```diff
@@ -11,291 +11,105 @@
-    
+model2netcdf.FATES <- function(outdir) {
+
     ## Load functions
-    mstmipvar <- PEcAn.utils::mstmipvar
     ncdim_def <- ncdf4::ncdim_def
     ncvar_def <- ncdf4::ncvar_def
     ncatt_get <- ncdf4::ncatt_get
     ncvar_add <- ncdf4::ncvar_add
-    misc.convert <- PEcAn.utils::misc.convert # unit conversions
+#    misc.convert <- PEcAn.utils::misc.convert # unit conversions
     logger.info <- PEcAn.utils::logger.info
     logger.severe <- PEcAn.utils::logger.severe
     logger.warn <- PEcAn.utils::logger.warn
+    
+    #        var_update("AR","AutoResp","kgC m-2 s-1")
+    var_update <- function(ncout,oldname,newname,newunits=NULL){
+      oldunits <- ncdf4::ncatt_get(ncin,oldname,"units")$value
+      if(is.null(newunits)) newunits = oldunits
+      newvar <- ncdf4::ncvar_def(name = newname, units = newunits, dim = xyt)
       
-    ## needs to be a generic function placed in utils, same goes for ED2 version of this function
-    ## also not all vars will be in every FATES run
-    getnetCDF <- function(nc, var) {
-      if (var %in% names(nc$var)) {
-        return(ncdf4::ncvar_get(nc, var))
+      if(is.null(ncout)) {
+        ncout <- ncdf4::nc_create(oname, newvar)
       } else {
-        logger.warn("Could not find", var, "in FATES output.")
-        return(-999)
+        ncdf4::ncvar_add(nc = ncout, v = newvar)
       }
+      try(ncdf4::nc_close(ncout))
+      ncout <- ncdf4::nc_open(oname,write=TRUE)
+      dat <- ncvar_get(ncin,oldname)
+      dat.new <- misc.convert(dat,oldunits,newunits)
+      ncdf4::ncvar_put(ncout,newname,array(dat.new,c(1,1,nt)))
+      return(ncout)
     }
-
+    
+    
     ## Get files and years
     files <- dir(outdir, "*clm2.h0.*.nc", full.names = TRUE)
     file.dates <- as.Date(sub(".nc", "", sub(".*clm2.h0.", "", files)))
-    file.years <- lubridate::year(file.dates)
-    num.years <- length(unique(file.years))
-    simulation.days <- as.Date(start_date):as.Date(end_date)
-    simulation.years <- as.numeric(unique(strftime(as.Date(simulation.days, origin = "1970-01-01"), "%Y")))
-    
-    # nescessary?  A check that output years and selected run years match
-    if (!all(file.years==simulation.years)) {
-      logger.severe("FATES output file years and simulation years don't match")
-    }
-    
+    years <- lubridate::year(file.dates)
+
     ## Loop over years
-    for (year in unique(file.years)) {
-        ysel <- which(file.years == year)  ## subselect files for selected year
-#        ysel <- sort(file.dates[ysel])  ## double check dates are in order
+    for (year in unique(years)) {
+        ysel <- which(years == year)  ## subselect files for selected year
         if (length(ysel) > 1) {
             logger.warn("PEcAn.FATES::model2netcdf.FATES does not currently support multiple files per year")
         }
         
-        # !!useful or do we not want this behavior? !!  Or some sort of clobber flag?
-        if (file.exists(file.path(outdir, paste(year, "nc", sep = ".")))) {
```

Comment:
In case we wanted to skip for any output already converted, but year I wasn't sure if this is what we wanted the behavior to be.  In this case we always re-create the output.